### ISSUE TITLE
Regression: SchemaError during version upgrade

### ISSUE THREAD

**Issue by nifgraup** [dfahlander/Dexie.js#1145]
**Steps**
Run this minimal repro https://jsfiddle.net/av6gw2re/,  same as the following:
```
const db = new Dexie("dbname");
db.version(1).stores({Y: "id"});
await db.open();
await db.close();

db.version(2).upgrade(trans => {
  console.log("Starting version 2 upgrade.")
  return trans.Y.count();
});
db.version(3).stores({
  Y: "id,X",
});
db.version(4).upgrade(trans => {
  console.log("Starting version 4 upgrade.")
  return trans.Y.where('X').equals('value').toArray();
});

try {
  await db.open();
} catch (e) {
  console.log(e);
} finally {
  await db.delete();
  console.log("All done.")
}
```
**What happens**
```
Starting version 2 upgrade.
Starting version 4 upgrade.
SchemaError: KeyPath X on object store Y is not indexed
All done.
```
**Expected**
No error as in Dexie 2:
```
Starting version 2 upgrade.
Starting version 4 upgrade.
All done.
```

**Comment by nifgraup** [comment]
Regression point 
```
commit 718f78980a033c7eac2ed3140c0dd52b9f170a4f
Author: David Fahlander <david.fahlander@gmail.com>
Date:   Fri Jun 1 15:14:25 2018 +0200

    Issue713 (#714)
    
    Making it possible to rename a table and also change primary key.
    
    Resolves #713
    Resolves #105
```

**Comment by dfahlander** [comment]
Thanks @nifgraup for the concise repro. The issue was a memoization in Transaction - if a table had been queried in an earlier upgrade, the old schema was reused later on.
